storage/grpc_client.go

```diff
@@ -216,8 +217,54 @@ func (c *grpcStorageClient) UpdateBucket(ctx context.Context, uattrs *BucketAttr
 func (c *grpcStorageClient) LockBucketRetentionPolicy(ctx context.Context, bucket string, conds *BucketConditions, opts ...storageOption) error {
 	return errMethodNotSupported
 }
-func (c *grpcStorageClient) ListObjects(ctx context.Context, bucket string, q *Query, opts ...storageOption) (*ObjectIterator, error) {
-	return nil, errMethodNotSupported
+func (c *grpcStorageClient) ListObjects(ctx context.Context, bucket string, q *Query, opts ...storageOption) *ObjectIterator {
+	s := callSettings(c.settings, opts...)
+	it := &ObjectIterator{
+		ctx: ctx,
+	}
+	if q != nil {
+		it.query = *q
+	}
+	req := &storagepb.ListObjectsRequest{
+		Parent:             bucketResourceName(globalProjectAlias, bucket),
+		Prefix:             it.query.Prefix,
+		Delimiter:          it.query.Delimiter,
+		Versions:           it.query.Versions,
+		LexicographicStart: it.query.StartOffset,
+		LexicographicEnd:   it.query.EndOffset,
+		// TODO(noahietz): Convert a projection to a FieldMask.
```

Comment:
Good call-out. I'm also curious if projection and FieldMask work exactly the same? It does make a difference performance-wise for JSON to set Projection to NoACL.

Reply:
No it doesn't work the same. We have to enumerate the fields that we *want* in a FieldMask. The existing "projection" implementation for JSON is almost like a "View" (https://google.aip.dev/157#view-enumeration). In FieldMask, `*` is like "full" projection. "NoACL" is gonna be a tad messier (have to enumerate every field _except_ the ACL field(s)).

Reply:
Ah yeah, I guess my main concern is whether the equivalent of NoACL will give the same performance advantage for gRPC.

Reply:
Oh Idk! That's a great question for the broader team.